MaxText/train.py

```diff
@@ -750,6 +752,14 @@ def train_loop(config, state=None):
   max_utils.close_summary_writer(writer)
   record_goodput(recorder, config, recorder.record_job_end_time if recorder else None)
   clear_buffered_metrics()
+  compiled = p_train_step.lower(state, example_batch, nextrng).compile()
```

Comment:
One of our internal tests actually hit a compile time OOM with this - I realize you need the wrap this with a mesh and logical axis rules context manager like we do the real p_train_step, e.g. here https://github.com/AI-Hypercomputer/maxtext/blob/27f8a81620d633ab2a7c67e6bab063d3964df474/MaxText/train.py#L683

Also do we need to print this every step - shouldn't this be roughly constant across iterations (maybe slightly different with real data due to token masks)?

Reply:
Thanks for the feedback. Addressed these two issues in the latest commit. Add the wrap and don't print every iteration.